flink-tests/src/test/java/org/apache/flink/api/connector/source/lib/IteratorSourcesITCase.java

```diff
@@ -52,6 +64,8 @@
                             .setNumberSlotsPerTaskManager(PARALLELISM)
                             .build());
 
+    @Rule public StreamCollector collector = new StreamCollector();
```

Comment:
Can we use `DataStream.executeAndCollect()` instead?
That is the main code users also use, let's give it as much exposure in tests as possible.